### PR TITLE
Add benchmarks for Microsoft.Extensions.Configuration.Xml

### PR DESCRIPTION
**Pull request by amoerie**
Hi

While creating [a pull request in the dotnet runtime repo](https://github.com/dotnet/runtime/pull/44608), I created some benchmarks to ensure my changes did not introduce a regression.

By request of @safern, I am adding these benchmarks here. 

I'm still struggling a bit with properly running them, and I'm short on time, so please give me a while to get this PR into a more polished state.

### CODE REVIEW

**Review by safern — APPROVED** [review]
Looks reasonable. I'd like @adamsitnik to take a look.

File: src/benchmarks/micro/libraries/Microsoft.Extensions.Configuration.Xml/TestFiles/deep.xml [hunk 1]
```diff
@@ -0,0 +1,104 @@
+﻿<settings>
```
Comment (danmoseley) [inline src/benchmarks/micro/libraries/Microsoft.Extensions.Configuration.Xml/TestFiles/deep.xml:1]: I just want to say this is beautiful.

**Comment by danmoseley** [comment]
Ack, this one is really old. @dotnet/area-extensions-configuration could one of you please sign off or review?

reopening to rebuild.

**Review by tarekgh — APPROVED** [review]

File: src/benchmarks/micro/libraries/Microsoft.Extensions.Configuration.Xml/XmlConfigurationProviderBenchmarks.cs [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+            fileStream.CopyTo(memoryStream);
+
+            return memoryStream;
+        }
+
+        [GlobalSetup]
+        public void GlobalSetup()
+        {
+            _simpleXml = ReadTestFile("simple.xml");
+            _deepXml = ReadTestFile("simple.xml");
```
Comment (amoerie) [inline src/benchmarks/micro/libraries/Microsoft.Extensions.Configuration.Xml/XmlConfigurationProviderBenchmarks.cs:31]: This is a copy paste error and should be a different file name. 🙈

**Comment by amoerie** [comment]
I completely forgot about this (looks like you guys did too 😁) but I can try to find some time this weekend or next week to tidy it up as I promised almost a year ago.

File: src/benchmarks/micro/libraries/Microsoft.Extensions.Configuration.Xml/TestFiles/names.xml [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        <DefaultConnection Name="44"><ConnectionString>TestConnectionString44</ConnectionString><Provider>SqlClient44</Provider></DefaultConnection>
+        <DefaultConnection Name="45"><ConnectionString>TestConnectionString45</ConnectionString><Provider>SqlClient45</Provider></DefaultConnection>
+        <DefaultConnection Name="46"><ConnectionString>TestConnectionString46</ConnectionString><Provider>SqlClient46</Provider></DefaultConnection>
+        <DefaultConnection Name="47"><ConnectionString>TestConnectionString47</ConnectionString><Provider>SqlClient47</Provider></DefaultConnection>
+        <DefaultConnection Name="48"><ConnectionString>TestConnectionString48</ConnectionString><Provider>SqlClient48</Provider></DefaultConnection>
+        <DefaultConnection Name="49"><ConnectionString>TestConnectionString49</ConnectionString><Provider>SqlClient49</Provider></DefaultConnection>
+        <DefaultConnection Name="50"><ConnectionString>TestConnectionString50</ConnectionString><Provider>SqlClient50</Provider></DefaultConnection>
+    </RepeatedElements>
+</settings>
+
```
Comment (maryamariyan) [inline src/benchmarks/micro/libraries/Microsoft.Extensions.Configuration.Xml/TestFiles/names.xml:55]: nit: extra blank line

**Review by maryamariyan — APPROVED** [review]
:shipit:

File: src/benchmarks/micro/libraries/Microsoft.Extensions.Configuration.Xml/TestFiles/names.xml (hunk 3 above)
Comment (amoerie) [reply src/benchmarks/micro/libraries/Microsoft.Extensions.Configuration.Xml/TestFiles/names.xml]: Fixed!

**Comment by amoerie** [comment]
No worries at all, this is probably super low in the priority list and rightly so.

Thanks for applying those fixes! They make a lot of sense.

I wanted to add a small note about the removal of repeated.xml:
The repeated.xml test file should work with the Microsoft.Extensions.Configuration.Xml package that came with .NET 6. (My PR added support for this, and that was the original context where I created these benchmarks)

So if the dependency on the Microsoft.Extensions.* libraries were updated to their respective 6.* versions, the benchmark should run.

**Comment by danmoseley** [comment]
Thanks @adamsitnik and @amoerie 

@amoerie if you see other benchmark gaps you are interested in contributing to, that would be welcome and you can expect us to merge them more promptly (this is unusual 😄 )

**Comment by amoerie** [comment]
> Thanks @adamsitnik and @amoerie 
> 
> @amoerie if you see other benchmark gaps you are interested in contributing to, that would be welcome and you can expect us to merge them more promptly (this is unusual 😄 )

I'm not exactly in the habit of perusing the dotnet repositories here, but I'll gladly contribute if I have an itch to scratch and I feel like I'm up to the task. 😅

Thanks for the review support, it is a very good learning resource!

**Pull request merged**